app/Http/Controllers/api/v1/RoleController.php

```diff
@@ -90,9 +90,7 @@ public function update(RoleRequest $request, Role $role)
 
         $user = Auth::user();
 
-        if (!($user->hasPermission('settings.manage')
-            && $user->hasPermission('roles.update')
-            && $user->hasPermission('roles.viewAny'))) {
+        if (!$user->hasPermission('roles.update')) {
```

Comment:
I would prefer to let the check how it is. Since this is more secure if something in the included permissions wouldn't work!